zrml/prediction-markets/src/orml_asset_registry.rs

```diff
@@ -0,0 +1,254 @@
+                                return Ok(());
+                            }
+                        }
+
+                        self.metadata.push((asset_id.clone(), meta));
+                        Ok(())
+                    }
+
+                    pub fn get_location(&self, asset_id: &$asset_id) -> Option<MultiLocation> {
+                        for (curr_id, meta) in &self.metadata {
+                            if curr_id == asset_id {
+                                return meta
+                                    .location
+                                    .as_ref()
+                                    .map(|versioned| versioned.clone().try_into().ok())
+                                    .flatten();
+                            }
+                        }
+
+                        None
+                    }
+
+                    pub fn get_asset_from_location(
+                        &self,
+                        location: &MultiLocation,
+                    ) -> Option<$asset_id> {
+                        for (curr_location, asset_id) in &self.location_to_asset {
+                            if curr_location == location {
+                                return Some(asset_id.clone());
+                            }
+                        }
+
+                        None
+                    }
+
+                    pub fn get_meta_from_location(
+                        &self,
+                        location: &MultiLocation,
+                    ) -> Option<AssetMetadata<$balance, $custom_metadata>> {
+                        let asset_id = self.get_asset_from_location(location)?;
+                        self.get_meta(&asset_id)
+                    }
+
+                    pub fn update_asset(
+                        &mut self,
+                        asset_id: $asset_id,
+                        _decimals: Option<u32>,
+                        _name: Option<Vec<u8>>,
+                        _symbol: Option<Vec<u8>>,
+                        _existential_deposit: Option<$balance>,
+                        _location: Option<Option<VersionedMultiLocation>>,
+                        _additional: Option<$custom_metadata>,
+                    ) -> DispatchResult {
+                        if let Some(_meta) = self.get_meta(&asset_id) {
+                            Ok(())
+                        } else {
+                            Err(DispatchError::Other("Asset not registered"))
+                        }
+                    }
+                }
+
+                impl RegistryState {
+                    fn new() -> Self {
+                        Self { location_to_asset: Vec::new(), metadata: Vec::new() }
+                    }
+                }
+
+                thread_local! {
+                    pub static STATE: RefCell<
+                        RegistryState,
+                    > = RefCell::new(RegistryState::new());
+                }
+            }
+        }
+    };
+}
```

Comment:
This is actually not necessary, we can achieve everything necessary with the asset registry pallet.
However, we still use the version `polkadot-v0.9.26` on asset registry, which unfortunately does not allow initializing genesis storage (we'd have to write raw keys and values...). So for now, this obviously is a won't fix. However, [from `polkadot-v0.9.29` onwards](https://github.com/open-web3-stack/open-runtime-module-library/compare/polkadot-v0.9.26...polkadot-v0.9.29#diff-973d213acaf59102ed6f6329665a78d77621db8e48f20758a6fdb9be6c75e675R114-R130), we can use the asset registry pallet for that. Can you create an issue to track that please?

Reply:
https://github.com/zeitgeistpm/zeitgeist/issues/936